src/components/Editor/Preview/index.js

```diff
@@ -72,52 +57,125 @@ class Preview extends PureComponent<Props, State> {
     this.updateHighlight(prevProps);
   }
 
+  componentWillUnmount() {
+    console.log("unmount");
+    if (this.disposalbles) {
+      this.disposalbles.forEach(dispose => dispose.dispose());
+      this.disposalbles = [];
+    }
+  }
+
   updateListeners(prevProps: ?Props) {
     const { isPaused } = this.props;
 
-    const { codeMirror } = this.props.editor;
-    const codeMirrorWrapper = codeMirror.getWrapperElement();
     const wasNotPaused = !prevProps || !prevProps.isPaused;
     const wasPaused = prevProps && prevProps.isPaused;
 
+    let lastTarget = null;
     if (isPaused && wasNotPaused) {
-      codeMirror.on("scroll", this.onScroll);
-      codeMirror.on("tokenenter", this.onTokenEnter);
-      codeMirror.on("tokenleave", this.onTokenLeave);
-      codeMirrorWrapper.addEventListener("mouseup", this.onMouseUp);
-      codeMirrorWrapper.addEventListener("mousedown", this.onMouseDown);
+      const { editor } = this.props;
+      this.disposalbles.push(
+        editor.monaco.onDidScrollChange(e => {
+          this.onScroll();
+        })
+      );
+      this.disposalbles.push(
+        editor.monaco.onMouseMove(e => {
+          if (
+            lastTarget &&
+            lastTarget.element === e.target.element &&
+            !this.hidden
+          ) {
+            return;
+          }
+          lastTarget = e.target;
+
+          // CONTENT_TEXT
+          if (e.target.type !== 6) {
+            this.onTokenLeave(e);
+            return;
+          }
+
+          const text = editor.monaco
+            .getModel()
+            .getWordAtPosition(e.target.position);
```

Comment:
this looks better than relying on the spans. i think we might want to use the cursor position if we can, but that might be fraught... This would help us differentiate in the case of hovering on `a`, `b`, or `c` e.g. `a.b.c.`